### DISCUSSION TITLE
how to reproduce loaded config (including hydra settings, overrides etc.) loading from .hydra dir ?

### DISCUSSION THREAD

**Discussion by YoelShoshan** [facebookresearch/hydra#2203] [Q&A]
Hi!

In .hydra dir we have 'config.yaml', 'hydra.yaml' and 'overrides.yaml'

I would like to load hydra config in the exact same way as it was loaded when the session was created (let's say, after that session was over). What is the recommended way to do it?

Motivation - machine learning scenario: 
Let's say that in the first run (that created this dir) I did training+validation, let's say that model selection is over 
and now I want to run test on a test-set, and I want to reproduce the exact config I previously had
(including things like the original hydra.run.dir )

**Comment by jieru-hu** [comment]
hi @YoelShoshan - there's a few options to achieve this:
1) we've recently added a experimental feature for rerunning in Hydra 1.2  check the doc here https://hydra.cc/docs/next/experimental/rerun/. Please try it out with the Hydra pre-release which you can install by `pip install hydra-core --pre --upgrade`
2) some other options available in https://github.com/facebookresearch/hydra/issues/1805#issuecomment-909362699
notice option 2 may not give you the exact same configs (OC resolvers might resolve to different values.)
